packages/playwright-core/src/utils/processLauncher.ts

```diff
@@ -116,34 +162,25 @@ export async function launchProcess(options: LaunchProcessOptions): Promise<Laun
   spawnedProcess.once('exit', (exitCode, signal) => {
     options.log(`[pid=${spawnedProcess.pid}] <process did exit: exitCode=${exitCode}, signal=${signal}>`);
     processClosed = true;
-    eventsHelper.removeEventListeners(listeners);
     gracefullyCloseSet.delete(gracefullyClose);
+    killSet.delete(killProcessAndCleanup);
```

Comment:
to retain old logic, we would need to remove event handlers if `killSet` and `gracefullyCloseSet` sets are empty.

Reply:
Let's try without this, and we'll bring it back if things break.